base/common/src/main/java/org/dogtagpki/common/CAInfo.java

```diff
@@ -93,10 +127,95 @@ public boolean equals(Object obj) {
+            archivalElement.appendChild(document.createTextNode(archivalMechanism));
+            infoElement.appendChild(archivalElement);
+        }
+
+        if (encryptAlgorithm != null) {
+            Element encryptElement = document.createElement("EncryptionAlgorithm");
+            encryptElement.appendChild(document.createTextNode(encryptAlgorithm));
+            infoElement.appendChild(encryptElement);
+        }
+
+        if (keyWrapAlgorithm != null) {
+            Element wrapElement = document.createElement("WrapAlgorithm");
+            wrapElement.appendChild(document.createTextNode(keyWrapAlgorithm));
+            infoElement.appendChild(wrapElement);
+        }
+
+        if (rsaPublicKeyWrapAlgorithm != null) {
+            Element rsaPublicWrapElement = document.createElement("RsaPublicKeyWrapAlgorithm");
+            rsaPublicWrapElement.appendChild(document.createTextNode(rsaPublicKeyWrapAlgorithm));
+            infoElement.appendChild(rsaPublicWrapElement);
+        }
+
+        if(caRsaPublicKeyWrapAlgorithm != null) {
+            Element caRsaPublicWrapElement = document.createElement("CaRsaPublicKeyWrapAlgorithm");
+            caRsaPublicWrapElement.appendChild(document.createTextNode(caRsaPublicKeyWrapAlgorithm));
+            infoElement.appendChild(caRsaPublicWrapElement);
+        }
+
+
+        return infoElement;
+    }
+
+    public static CAInfo fromDOM(Element infoElement) {
+
+        CAInfo info = new CAInfo();
+
+        fromDOM(infoElement, info);
+
+        NodeList archivalList = infoElement.getElementsByTagName("ArchivalMechanism");
+        if (archivalList.getLength() > 0) {
+            String value = archivalList.item(0).getTextContent();
+            info.setArchivalMechanism(value);
+        }
+
+        NodeList encryptionList = infoElement.getElementsByTagName("EncryptionAlgorithm");
+        if (encryptionList.getLength() > 0) {
+            String value = encryptionList.item(0).getTextContent();
+            info.setEncryptAlgorithm(value);
+        }
+
+        NodeList wrapList = infoElement.getElementsByTagName("KeyWrapAlgorithm");
+        if (wrapList.getLength() > 0) {
+            String value = wrapList.item(0).getTextContent();
+            info.setKeyWrapAlgorithm(value);
+        }
+
+        NodeList rsaPublicKeyWrapList = infoElement.getElementsByTagName("RsaPublicKeyWrapAlgorithm");
+        if (rsaPublicKeyWrapList.getLength() > 0) {
+            String value = rsaPublicKeyWrapList.item(0).getTextContent();
+            info.setRsaPublicKeyWrapAlgorithm(value);
+        }
+
+        NodeList caRsaPublicKeyWrapList = infoElement.getElementsByTagName("CaRsaPublicKeyWrapAlgorithm");
+        if (caRsaPublicKeyWrapList.getLength() > 0) {
+            String value = caRsaPublicKeyWrapList.item(0).getTextContent();
+            info.setCaRsaPublicKeyWrapAlgorithm(value);
+        }
+
+        return info;
+    }
```

Comment:
I have never used Jackson databind but JAXB and JAX-RS so I need to investigate more but I am wondering why this mapping is not done with an object mapper but manually. Is there a specific reason?

Reply:
Classes serialised with Jackson should not use their own methods to go to/from JSON, I created the [JSONSerializer](https://github.com/dogtagpki/pki/blob/master/base/common/src/main/java/com/netscape/certsrv/util/JSONSerializer.java) interface with default methods so that all classes that need to do this can simply implement `JSONSerializer` rather than having tons of boilerplate code. I didn't bother making an equivalent `XMLSerializer` interface because we are removing XML, hopefully in PKI 12. We could use an `ObjectMapper` here but my question is do we need this XML code at all, can we just rely on the JSON?

Reply:
I do not think this XML is needed but I would keep for the moment and remove when the super class `com.netscape.certsrv.base.RESTMessage.java` will drop the XML support.
However, if you already have started on removing the XML for other messagges then we can follow the approach here (in the master for the porting).

Reply:
For the moment, we have tried to provide JSON equivalents to all XML serialisations (there are numerous PRs of prior art), which I think we have now provided, so usage of XML is considered deprecated in PKI 11 (hence my reluctance to add more).

I see the point of keeping the XML though as the superclass allows it, so we will just remove it later for PKI 12 along with all the others. There are currently 94 `toDOM` methods so adding a 95th is not going to add too much work I guess.

I will retract my request for changes and approve, merging is fine by me :+1: